src/robot/ConcreteRobot.cpp

```diff
@@ -331,6 +331,64 @@ ConcreteRobot::getTrajectoryPostProcessor(
       feasibilityApproxTolerance);
 }
 
+//==============================================================================
+std::shared_ptr<aikido::planner::TrajectoryPostProcessor>
+ConcreteRobot::getTrajectoryPostProcessor(
+    const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+    const aikido::planner::PostProcessorParams& postProcessorParams) const
+{
+  // TODO (sniyaz):Should we  delete `smoothPath`, `retimePath`, and
+  // `retimePathWithKunz` in a subsequent PR?
+  // TODO: (sniyaz): Include custom limits in params.
+  switch (postProcessorParams.mPostProcessorType)
+  {
+    case aikido::planner::PostProcessorType::HAUSER:
+      return std::make_shared<ParabolicSmoother>(
+          getVelocityLimits(*metaSkeleton),
+          getAccelerationLimits(*metaSkeleton),
+          postProcessorParams.mHauserParams.mEnableShortcut,
+          postProcessorParams.mHauserParams.mEnableBlend,
+          postProcessorParams.mHauserParams.mShortcutTimelimit,
+          postProcessorParams.mHauserParams.mBlendRadius,
+          postProcessorParams.mHauserParams.mBlendIterations,
+          postProcessorParams.mHauserParams.mFeasibilityCheckResolution,
+          postProcessorParams.mHauserParams.mFeasibilityApproxTolerance);
+    case aikido::planner::PostProcessorType::KUNZ:
+      return std::make_shared<KunzRetimer>(
+          getVelocityLimits(*metaSkeleton),
+          getAccelerationLimits(*metaSkeleton),
+          postProcessorParams.mKunzParams.mMaxDeviation,
+          postProcessorParams.mKunzParams.mTimeStep);
+    default:
+      // TODO (sniyaz): Can this be made a compile-time check instead?
```

Comment:
This can be done with templates, see the full review.

Reply:
Gotcha!